11.3 RNs (#10132)
mendix/docs: content/en/docs/releasenotes/studio-pro/11/11.3.md

---
title: "11.3"
url: /releasenotes/studio-pro/11.3/
description: "The release notes for Mendix Studio Pro 11.3 (including all patches) with details on new features, bug fixes, and known issues."
weight: 98
---

## 11.3.0

**Release date: September _____, 2025**

{{% button color="info" href="https://marketplace.mendix.com/link/studiopro/11.3.0" text="Go to Marketplace" title="Download version 11.3.0 from the Marketplace" %}}

### New Features
    
* We introduced a setting in the **Advanced** tab of the **Preferences** dialog to enable starting Studio Pro in Extension Development Mode, so users do not need to provide the feature flag themselves.
* We introduced a new API for showing version control information (the system, the branch name, and last commit) from web extensions. It is available under `studioPro.ui.versionControl` in the web extensibility API.
* The experimental Java API `OqlStatement` to run OQL statements now also supports [UPDATE statements](/refguide/oql-statements/#oql-update) to update attributes.
* It is now possible to open Studio Pro from the Select App window without selecting an app, which allows accessing menu items like the **Preferences** dialog.

> * The External Database connector now supports `Dynamic queries`, allowing users to modify their queries at runtime using variables or parameters. 

Author: @quinntracy doc update needed?

- We introduced a new API for showing version control information(the system, the branch name and last commit) from web extensions. It is available under `studioPro.ui.versionControl` in the web extensibility API.
- The experimental Java API `OqlStatement` to run OQL statements now also supports UPDATE statements to update attributes.
- It is now possible to open Studio Pro from the Select App window without selecting an app, which allows accessing tools like full Preferences dialog or opening the Logs directory.

### Improvements
    

* We upgraded the JDBC driver for Microsoft SQL Server to version 13.2.0.
    * If you connect to SQL Server using integrated security in a module such as [Database Connector](https://marketplace.mendix.com/link/component/2888), you may need to update the JDBC driver in `userlib` as well.
- Maia for Pages now has knowledge about styling properties and can set and update their values.
- When either the `X-Forwarded-Proto` or `X-Forwarded-Scheme` header is set to `http` cookies will now be set without the `secure` attribute, even if the `ApplicationRootUrl` is set to a secure URL.

> - In React Client, we now offer a quick fix for finding the usages of a select page configured for an association widget.

Author: I'll look into documenting this quick fix more explicitly.

- We added support for converting Data grid columns that are configured over reference sets.
- We have made several improvements to the modernized app explorer to enhance usability, performance, and stability. We have (temporarily) disabled the [Enter], [Left Arrow], and [Right Arrow] hotkeys on expandable nodes and [Left Arrow] on leaf nodes in the modernized app explorer.
- We added `Check all` and `Uncheck all` functionality while selecting REST requests, when importing an OpenApi contract in Consumed Rest Services.
- When creating a new branch, invalid characters (~ ^ : ? * [ \) are now automatically replaced by a dash (-) to adhere to Git's naming limitations.

> - Studio pro now replaces the invalid characters in branch name dialog with dashes as they are typed.

Author: Are both these dash bullets valid, or duplications of each other?

Reviewer: I've made a suggestion for combining them.

- We fixed an issue in workflow events, where a second UserTaskEnded was sent when the workflow was restarted. It now sends only one UserTaskEnded event.
- We fixed an issue in workflow user tasks, where the assigned users were not taken into account in determining the number of users that can complete the task.
- We improved the Japanese and Chinese translations based on the feedback from our users. (Ticket 258099)
- We fixed an issue where database synchronization would sometimes unnecessarily recreate the uniqueness constraint on the primary key mapping table of an external entity. (Ticket 255223)
- We fixed an issue where using an Oracle database that uses a localized character set would result in an error when inserting or retrieving string values due to missing conversion library. (Ticket 257140)
- We fixed an issue where database synchronization would fail when all of the following applied in the same deployment:  (Ticket 258046)
  * An association in the database contained any dangling references
  * Foreign key constraints were newly enabled
  * The association storage was changed to or from being a direct association 

> - We fixed an issue where a foreign key constraint violation in the database could result in a NullPointerException. (Ticket 258430)

Author: The thing that broke 10.24.7

Author: Maybe I should tweak the lang to be 1-1 as the other bullet.

Reviewer: @ConnorLand was this ever released in 11.x? If not there was nothing to fix and we should get rid of this, otherwise people will worry that there was a bug in 11.2?

---
title: "11.3"
url: /releasenotes/studio-pro/11.3/
description: "The release notes for Mendix Studio Pro 11.3 (including all patches) with details on new features, bug fixes, and known issues."
weight: 98
---

## 11.3.0

**Release date: September _____, 2025**

{{% button color="info" href="https://marketplace.mendix.com/link/studiopro/11.3.0" text="Go to Marketplace" title="Download version 11.3.0 from the Marketplace" %}}

### New Features
    

> * We introduced a settings in the Preferences form to enable starting Studio Pro in Extension Development Mode under the Advanced tab, so users do not need to provide the feature flag themselves.

Reviewer: ```suggestion
* We introduced a setting in the **Advanced** tab of the **Preferences** dialog to enable starting Studio Pro in Extension Development Mode, so users do not need to provide the feature flag themselves.
```

> * We introduced a new API for showing version control information(the system, the branch name and last commit) from web extensions. It is available under `studioPro.ui.versionControl` in the web extensibility API.

Reviewer: ```suggestion
* We introduced a new API for showing version control information (the system, the branch name, and last commit) from web extensions. It is available under `studioPro.ui.versionControl` in the web extensibility API.
```
Not sure what it means by "the system" though. Do you want me to follow up?

Author: No, your commit is enough.

> * The experimental Java API `OqlStatement` to run OQL statements now also supports UPDATE statements to update attributes.

Reviewer: ```suggestion
* The experimental Java API `OqlStatement` to run OQL statements now also supports [UPDATE statements](/refguide/oql-statements/#oql-update) to update attributes.
```

> * It is now possible to open Studio Pro from the Select App window without selecting an app, which allows accessing tools like full Preferences dialog or opening the Logs directory.

Reviewer: What does "full Preferences" mean? And what is the "logs directory"? I think we can probably say

```suggestion
* It is now possible to open Studio Pro from the Select App window without selecting an app, which allows accessing menu items like the Preferences dialog.
```

### Improvements

* We upgraded the JDBC driver for Microsoft SQL Server to version 13.2.0.
    * If you connect to SQL Server using integrated security in a module such as [Database Connector](https://marketplace.mendix.com/link/component/2888), you may need to update the JDBC driver in `userlib` as well.
* Maia for Pages now has knowledge about styling properties and can set and update their values.

> * When either the `X-Forwarded-Proto` or `X-Forwarded-Scheme` header is set to `http` cookies will now be set without the `secure` attribute, even if the `ApplicationRootUrl` is set to a secure URL.

Reviewer: ```suggestion
* When either the `X-Forwarded-Proto` or `X-Forwarded-Scheme` header is set to `http`, cookies will now be set without the `secure` attribute, even if the `ApplicationRootUrl` is set to a secure URL.
```

> * In React Client, we now offer a quick fix for finding the usages of a select page configured for an association widget.

Reviewer: Should this be "**the** React Client"?

* We added support for converting Data grid columns that are configured over reference sets.
* We have made several improvements to the modernized app explorer to enhance usability, performance, and stability. We have (temporarily) disabled the [Enter], [Left Arrow], and [Right Arrow] hotkeys on expandable nodes and [Left Arrow] on leaf nodes in the modernized app explorer.
* We added `Check all` and `Uncheck all` functionality while selecting REST requests, when importing an OpenApi contract in Consumed Rest Services.
* When creating a new branch, invalid characters (~ ^ : ? * [ \) are now automatically replaced by a dash (-) to adhere to Git's naming limitations.

> * Studio pro now replaces the invalid characters in branch name dialog with dashes as they are typed.

Reviewer: I think these two are so similar they can be combined.
```suggestion
* When creating a new branch, invalid characters (~ ^ : ? * [ \) are now automatically replaced by a dash (-) to adhere to Git's naming limitations. This means that Studio Pro now replaces the invalid characters in branch name dialog with dashes as they are typed.
```

* We have added a new `com.mendix.webui.AuthTokenTimeout` runtime setting as an alias for the existing `com.mendix.webui.HybridAppLoginTimeOut` setting. This was done to avoid confusion based on `com.mendix.webui.HybridAppLoginTimeOut` mentioning 'hybrid', a previously removed technology. However, the older setting can still be used for backwards compatibility.

### Fixes

> * We fixed issue in published OData services where requests that filtered based on associated data combined with 'and' would receive a response containing data that did not match the filter. (Ticket 241825)

Reviewer: ```suggestion
* We fixed an issue in published OData services where requests that filtered based on associated data combined with 'and' would receive a response containing data that did not match the filter. (Ticket 241825)
```

* We fixed a rare issue where an error pop-up window could appear if a user attempted a version control operation while Git optimization was running for an extended period of time. (Ticket 246741)

> * We fixed an issue for Import Mappings and Export Mappings that caused checkboxes in the **Select schema elements** tree structure to be set incorrectly. This happened for XML schema inheritance and choice elements. (Ticket 249795)

Reviewer: ```suggestion
* We fixed an issue with Import Mappings and Export Mappings that caused checkboxes in the **Select schema elements** tree structure to be set incorrectly. This happened for XML schema inheritance and choice elements. (Ticket 249795)
```

* We fixed an offline native mobile issue that caused a JSON parse exception (unexpected character) during synchronization. (Ticket 250114, Ticket 256696)
* We fixed an issue that could result in the error *Duplicate objects are not allowed* in the console log when synchronizing offline. Please note that the root cause of this issue is the use of background synchronization, which is [not recommended for offline apps](/refguide/synchronize/#background-sync). (Ticket 235130, 251202)
* We fixed an issue where an entity with **Synchronization Config** set to **Never** was removed from the list view or Data grid 2 when **Synchronize All** was triggered (due to excessive cache deletion). For more information on synchronization, see [Offline Synchronization](/refguide/mobile/building-efficient-mobile-apps/offlinefirst-data/synchronization/). (Ticket 253629)

> * We fixed an issue where retrieving from database in nanoflows failed when using many-to-many associations with XPath constraint going over an association. (Ticket 253996)

Reviewer: ```suggestion
* We fixed an issue where retrieving from the database in nanoflows failed when using many-to-many associations with an XPath constraint going over an association. (Ticket 253996)
```

> * We fixed an issue in offline apps causing reference sets (n:m associations) incorrectly not showing related data with the database data source. (Ticket 254407)

Reviewer: ```suggestion
* We fixed an issue in offline apps causing reference sets (n:m associations) to incorrectly not show related data with the database data source. (Ticket 254407)
```

> * We fixed an issue where an **Oops** pop-up window appeared during deployment if multiple pages using the same snippet are processed simultaneously. (Ticket 254648)

Reviewer: ```suggestion
* We fixed an issue where an **Oops** pop-up window appeared during deployment if multiple pages using the same snippet were processed simultaneously. (Ticket 254648)
```

> * We fixed an issue where entering only whitespace to the JDK location setting causes Studio Pro fail to start. (Ticket 254761)

Reviewer: ```suggestion
* We fixed an issue where entering only whitespace to the JDK location setting caused Studio Pro to fail to start. (Ticket 254761)
```

Author: Good tense catches.

* We fixed an issue where an expression could not be set when the number of available variables was too large, causing the variable list to expand and block access to the expression editor. (Ticket 254957)

> * We now allow filtering over reference sets for database data sources, previously allowed only for list views. (Ticket 251833)

Reviewer: ```suggestion
* We now allow filtering over reference sets for database data sources. This was previously allowed only for list views. (Ticket 251833)
```

* We fixed an issue where database synchronization sometimes unnecessarily recreated the uniqueness constraint on the primary key mapping table of an external entity. (Ticket 255223)
* We fixed an issue where an **Oops** pop-up window was displayed when the page variables button was clicked in structure mode for a page template. (Ticket 255394)

> * We fixed an issue where the combo box widget failed to render when the **Store** value is configured with a page or snippet variable. (Ticket 256493)

Reviewer: ```suggestion
* We fixed an issue where the combo box widget failed to render when the **Store** value was configured with a page or snippet variable. (Ticket 256493)
```

* We fixed the **Tab Container** widget to avoid loading hidden tab content when the conditional visibility expression refers to a page parameter. (Ticket 256717)
* We fixed an issue where using an Oracle database that uses a localized character set would result in an error when inserting or retrieving string values due to missing conversion library. (Ticket 257140)
* We fixed an error where apps would attempt to clear their local cache while developing locally when not running in a secure context (such as **localhost**). (Ticket 257026)

> * We fixed an issue where doing a hot reload while using the React client would lead the to deployment spinner spinning infinitely in some projects. (Tickets 257026, 257859)

Reviewer: ```suggestion
* We fixed an issue where doing a hot reload while using the React client would lead to the deployment spinner spinning for ever in some projects. (Tickets 257026, 257859)
```

> * We fixed an issue on offline-associated entities pertaining to `startsWith`, which is not a function error in synchronize actions. (Ticket 257346)

Reviewer: This isn't completely clear to me, but I never work with offline stuff so if it makes sense to you that's fine.

Author: Yeah, this clears the bar. But I hear you, lol.

* We fixed an issue where database synchronization would fail when all of the following occurred during the same deployment: (Ticket 258046)

    * An association in the database contained any dangling references
    * Foreign key constraints were newly enabled
    * The association storage was changed to or from being a direct association 

* We improved the Japanese and Chinese translations based on the feedback from our users. (Ticket 258099)
* We fixed an issue where a foreign key constraint violation in the database could result in a `NullPointerException`. (Ticket 258430)
* We fixed an issue where SVG images would load with incorrect dimensions. (Ticket 258901)
* We fixed an issue where passing an empty enumeration value to a microflow executed in a task queue would cause the execution to fail. (Ticket 259170)

> * We fixed an issue in the System.WorkflowEndedUserTask where the state did not the match the activity records.

Reviewer: ```suggestion
* We fixed an issue with `System.WorkflowEndedUserTask` where the state did not the match the activity records.
```

> * We fixed an issue in workflow events, where a second UserTaskEnded was sent when the workflow was restarted. It now sends only one UserTaskEnded event.

Reviewer: ```suggestion
* We fixed an issue in workflow events, where a second `UserTaskEnded` was sent when the workflow was restarted. It now sends only one `UserTaskEnded` event.
```

* We fixed an issue in workflow user tasks, where the assigned users were not taken into account in determining the number of users that can complete the task.
* We fixed an issue where database synchronization could fail on SQL Server when a specialization of `System.User` was removed and there was still an `owner` or `changedBy` reference from an instance of `System.User` or a different specialization of it to an instance of the specialization being removed. 
* We fixed an issue with copy/pasting in tab containers in design mode. It is now possible to paste into the container of the tab as well as pasting directly on the tab header this also affects structure mode.
* We fixed an issue where data grid action buttons with empty parameters would cause runtime errors when calling microflows.

> * We fixed an issue where an **Oops** pop-up window was shown during conversion when a select page of the converted reference selector does not contain a data container widget.

Reviewer: Does this mean "a select**ed** page" or is it some special kind of page called a _select_ page?

Author: It should be selected page, for sure. Nice catch. I'll change it.

* We fixed an issue where an **Oops** pop-up window was displayed when switching the data source type of a combo box.

> * We fixed an issue when using file dialogs and the persisted file/directory/drive do not exist anymore.

Reviewer: ```suggestion
* We fixed an issue when using file dialogs and the persisted file/directory/drive does not exist anymore.
```

Author: Ah, but we must also eliminate the slashes. This RN exists in another patch. I'll copy paste the improved version.

> * We have upgraded React Native version to 0.77.3. This addresses an issue that caused the app to stop working after reload.

Reviewer: Should this be higher up the list of fixes? It seems quite an important thing to know about if you are a React programmer?

Author: Good idea. I've only ranked the ticketed fixes, so I'll rank non-ticketed fixes Monday and put this one high (as well as other big ones).

* We fixed an issue for Export Mappings where it was incorrectly possible to unselect the Message Definition root element in the 'Select schema elements' tree structure.
* We fixed an issue for Export Mappings where a user would uncheck a checkbox in the 'Select schema elements' tree structure, but not all children elements were unchecked.

> * We fixed an issue for Import Mappings and Export Mappings where a user would check a checkbox in the 'Select schema elements' tree structure 5 levels deep or more and the tree failed to expand the tree to show children items. This made it possible to not select required schema elements.

Reviewer: Possibly clearer to say
```suggestion
* We fixed an issue for Import Mappings and Export Mappings where a user would check a checkbox in the 'Select schema elements' tree structure five levels deep or more and the tree failed to expand the tree to show children items. This could mean that not all required schema elements were selected.
```

> * We fixed an issue for Export Mappings where XSD extension attributes checkboxes in the 'Select schema elements' tree structure were incorrectly not selected when the user selected the checkbox of their parent attribute.

Reviewer: All these seem very closely related. Should they be merged?
Perhaps @quinntracy would know?

> * We fixed an issue in Published Rest Service where wrong response schema was being generated with Export Mapping for OpenAPI v3.

Reviewer: ```suggestion
* We fixed an issue in Published Rest Service where the wrong response schema was being generated with Export Mapping for OpenAPI v3.
```

> * We fixed an issue in Create Object Activity which would fail when creating an instance of a remote non-persistable entity in a microflow.

Reviewer: ```suggestion
* We fixed an issue in Create Object Activity where it would fail when creating an instance of a remote non-persistable entity in a microflow.
```

Author: Good, and I will also correct it to **Create object activity**.

* We fixed an issue in consumed REST services where choosing an OpenAPI contract without any operations would result in an error.
* We fixed an issue in the Consumed REST Service document in Studio Pro where clicking OK on a dialog would switch the focus away from Studio Pro.
* We fixed an issue where changing MxAdmin user's name in StudioPro does not result in changing MxAdmin user's name in database. (Ticket 255505)

> * We fixed an issue where a warning was logged on startup about an already registered Jetty gauge when multiple listening addresses are specified. The connection metrics for Jetty are added per listening address and they now contain a tag 'port' with the port number and a tag 'host' if a host is specified. This is applicable to the metrics `jetty.connections.bytes.in`, `jetty.connections.bytes.out`, `jetty.connections.current`, `jetty.connections.max`, `jetty.connections.messages.in`, `jetty.connections.messages.out` and `jetty.connections.request`. Be aware of the fact that there are usually two listening addresses when you specify 'Open to current machine' for Runtime port security, e.g. 127.0.0.1 and ::1.

Reviewer: ```suggestion
* We fixed an issue where a warning was logged on startup about an already registered Jetty gauge when multiple listening addresses are specified. The connection metrics for Jetty are added per listening address and they now contain a tag 'port' with the port number and a tag 'host' if a host is specified. This is applicable to the following metrics: 

    * `jetty.connections.bytes.in`
    * `jetty.connections.bytes.out`
    * `jetty.connections.current`
    * `jetty.connections.max`
    * `jetty.connections.messages.in`
    * `jetty.connections.messages.out`
    * `jetty.connections.request`
    
    Be aware of the fact that there are usually two listening addresses when you specify 'Open to current machine' for Runtime port security, e.g. 127.0.0.1 and ::1.
```

> * As Git is the only supported repository-type at present, the repository-type selector has been removed from the 'Import Solution App Package' popup window when 'New Mendix Team Server' is selected.

Reviewer: ```suggestion
* As Git is now the only supported repository-type, the repository-type selector has been removed from the 'Import Solution App Package' popup window when 'New Mendix Team Server' is selected.
```

Author: Ah, and we should delete the hyphen between repository and type. I'll do that.

* We fixed the issue when Push button is disabled after opening an app with local commits while auto-fetch is disabled.
* We fixed an issue where a user is asked to provide git credentials for Repository optimization.

> * We fixed an issue where user gets an **Oops** pop-up window when creating non-versioned app.

Reviewer: ```suggestion
* We fixed an issue where user gets an **Oops** pop-up window when creating a non-versioned app.
```

> * We fixed an issue with the icon of the security level dialog that the the dark-mode icon was visible in light-mode.

Reviewer: ```suggestion
* We fixed an issue with the icon of the security level dialog where the dark-mode icon was visible in light-mode.
```

> * We fixed an issue where a "404 - file not found for file: .well-known%2Fappspecific%2Fcom.chrome.devtools.json" error was shown in the Studio Pro console when opening a React app in Chrome with dev tools open. This error was caused by Chrome's automatic workspace folders feature.

Reviewer: I think this %2F is a slash and this should be:
```suggestion
* We fixed an issue where a "404 - file not found for file: .well-known/appspecific/com.chrome.devtools.json" error was shown in the Studio Pro console when opening a React app in Chrome with dev tools open. This error was caused by Chrome's automatic workspace folders feature.
```

But probably best to check with the relevant team

Author: Oh right. Weird. I wonder if GitLab or something changed it.

> * Fixed an issue with the client parser API where the `groups` option would also require the `places` option.

Reviewer: ```suggestion
* We fixed an issue with the client parser API where the `groups` option would also require the `places` option.
```

* We fixed an issue where previous pages would not load in the React client after closing a page that was opened from a popup.
* We fixed an issue where a "Manifest: unknown 'display' value ignored" warning was shown in the browser console when opening a non-PWA React app.

> * We fixed an issue where pages with a mix of persistent and non-persistent entity parameters did not load properly on the first time.

Reviewer: ```suggestion
* We fixed an issue where pages with a mix of persistent and non-persistent entity parameters did not load properly the first time.
```
